coordinator/coordinator.go

```diff
@@ -743,7 +743,14 @@ func (c *Coordinator) runFeatureMaterializeJob(resID metadata.ResourceID, schedu
 		}
 	}(sourceStore)
 
-	featureProvider, err := feature.FetchProvider(c.Metadata, context.Background())
+	var featureProvider *metadata.Provider
+	if feature.Provider() != "" {
+		featureProvider, err = feature.FetchProvider(c.Metadata, context.Background())
+		if err != nil {
+			return fmt.Errorf("could not fetch online provider: %v", err)
+		}
+	}
+
 	if err != nil {
```

Comment:
do we need to remove this check?

Reply:
why would we need to remove it?

Reply:
you are performing the same check in the if conditional above?